recogym/envs/abstract.py

```diff
@@ -102,11 +102,21 @@ def reset(self, user_id=0):
         # Record number of times each product seen for static policy calculation.
         self.organic_views = np.zeros(self.config.num_products)
 
-    def generate_organic_sessions(self):
+    def generate_organic_sessions(self, initial_product = None):
 
         # Initialize session.
         session = OrganicSessions()
 
+        if initial_product is not None :
+            # If a product is specified, use it as initial page for the organic session
+            session.next(
+                DefaultContext(self.current_time, self.current_user_id),
+                initial_product
+            )
+
+            # Update markov state.
+            self.update_state()
```

Comment:
Time is not updated, thus BOTH events
  Bandit (with Action ID = initial_product)
  Organic (with the Product ID = initial_product)
will happen at the same time.

In the case when `initial_product is not None' and we update the state, the state could not be 'organic', that means that we could not generate pure/real Organic events (we do not enter to the following while loop).

Reply:
Time is updated in update_state. For the first organic event after a bandit event, it will be updated in the update_state command of the step function, just after a click has been drawn. Thus we won't encounter the problem of having a bandit event and an organic event at the same time.

We are in the organic state even though the initial_product is not None (we just choose how to start the organic session), thus the state update works as usual, and we are able to generate organic events, with the only difference that we can choose the initial page of the organic session, which could be useful just after a click.